apps/extension/src/App/App.tsx

```diff
@@ -1,57 +1,88 @@
 import React, { useEffect, useState } from "react";
-import { BrowserRouter, Routes, Route } from "react-router-dom";
 import { ThemeProvider } from "styled-components";
-import browser from "webextension-polyfill";
+import { Routes, Route, useNavigate } from "react-router-dom";
 
-import { Button, ButtonVariant } from "@anoma/components";
 import { getTheme } from "@anoma/utils";
-
 import { ExtensionRequester } from "extension";
 import { Ports } from "router";
-import { DerivedAccount, QueryAccountsMsg } from "background/keyring";
+import {
+  CheckIsLockedMsg,
+  DerivedAccount,
+  QueryAccountsMsg,
+} from "background/keyring";
+
 import {
   AppContainer,
   BottomSection,
   ContentContainer,
-  LoadingError,
   GlobalStyles,
   TopSection,
 } from "./App.components";
-import Accounts from "./Accounts/Accounts";
-import AddAccount from "./Accounts/AddAccount";
-import { TopLevelRoute } from "App/types";
+import { LockWrapper } from "./LockWrapper";
+import { Accounts, AddAccount } from "./Accounts";
+import { Login } from "./Login";
+import { Setup } from "./Setup";
+import { TopLevelRoute } from "./types";
+import { Loading } from "./Loading";
 
 const requester = new ExtensionRequester();
 
-enum Status {
+export enum Status {
   Completed,
   Pending,
   Failed,
 }
 
 export const App: React.FC = () => {
   const theme = getTheme(true, false);
+  const navigate = useNavigate();
   const [status, setStatus] = useState<Status>();
```

Comment:
nit: This went in already earlier but now that it is being used outside of this file, I feel now that it is being used outside of this file, this name is even more ambiguous as one cannot easily read the code to get the whole context of "what status this is".

```suggestion
  const [accountsLoadingStatus, setAccountsLoadingStatus] = useState<Status>();
```

Reply:
Since these are pretty quick changes, I'll add them in my current PR (https://github.com/anoma/namada-interface/pull/114) to avoid potential conflicts.